src/app/components/generic-event-page/generic-event-page.component.ts

```diff
@@ -57,20 +77,42 @@ export class GenericEventPageComponent implements OnInit {
   }
 
   filterEvents(filterData): void{
-    this.filterDataService.getFilteredEvents(filterData).subscribe((result) => {
-      this.eventsArray = result.result;
-      if(this.sortData == null){
-        this.eventsArray = this.sortDataService.sort(this.defaultSortData, this.eventsArray);
-      }
-      else{
-        this.eventsArray = this.sortDataService.sort(this.sortData, this.eventsArray);
-      }
-    });
+    
+    filterData = this.prepareFilterData(filterData);
+
+    if(this.title === 'Joined Events' || this.title === 'Hosted Events'){
+      this.filterDataService.getNotEndedEvents(filterData).subscribe((result) => {
+        this.eventsArray = result.result;
+        if(this.sortData == null){
+          this.eventsArray = this.sortDataService.sort(this.defaultSortData, this.eventsArray);
+        }
+        else{
+          this.eventsArray = this.sortDataService.sort(this.sortData, this.eventsArray);
+        }
+      });
+    }
+    else if(this.title === 'Future Events'){
+      this.filterDataService.getFilteredEvents(filterData).subscribe((result) => {
+        this.eventsArray = result.result;
+        if(this.sortData == null){
+          this.eventsArray = this.sortDataService.sort(this.defaultSortData, this.eventsArray);
+        }
+        else{
+          this.eventsArray = this.sortDataService.sort(this.sortData, this.eventsArray);
+        }
```

Comment:
this.eventsArray = !this.sortData ? this.sortDataService.sort(this.defaultSortData, this.eventsArray) : this.sortDataService.sort(this.sortData, this.eventsArray);